lasagne/layers/recurrent.py

```diff
@@ -58,70 +97,47 @@ def __init__(self, input_layer, input_to_hidden, hidden_to_hidden,
 
         # Get the batch size and number of units based on the expected output
         # of the input-to-hidden layer
-        (n_batch, self.num_units) = self.input_to_hidden.get_output_shape()
+        (self.n_batch, _, self.num_inputs) = self.input_shape
+        self.num_units = i2h_out[-1]
 
         # Initialize hidden state
-        self.hid_init = self.create_param(hid_init, (n_batch, self.num_units))
-
-    def get_params(self):
-        '''
-        Get all parameters of this layer.
+        self.hid_init = self.add_param(hid_init, (1, self.num_units),
+                                       trainable=learn_init, name="hid_init")
 
-        :returns:
-            - params : list of theano.shared
-                List of all parameters
-        '''
-        params = (helper.get_all_params(self.input_to_hidden) +
-                  helper.get_all_params(self.hidden_to_hidden))
+    def get_params(self, **tags):
+        params = (helper.get_all_params(self.input_to_hidden, **tags) +
+                  helper.get_all_params(self.hidden_to_hidden, **tags))
 
         if self.learn_init:
```

Comment:
If self.hid_init has the trainable flag set (see line 105), can’t we omit this part now?